lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
```

Comment:
I think it can happen in cases, where the exception is e.g. a "dns-exception" before the actual http exchange happens.